src/JsonSchemaMapper/JsonSchemaMapper.ReflectionHelpers.cs

```diff
@@ -27,7 +27,7 @@ static partial class JsonSchemaMapper
     private static Type GetElementType(JsonTypeInfo typeInfo)
     {
         Debug.Assert(typeInfo.Kind is JsonTypeInfoKind.Enumerable or JsonTypeInfoKind.Dictionary);
-        return (Type)typeof(JsonTypeInfo).GetProperty("ElementType", BindingFlags.Instance | BindingFlags.NonPublic)?.GetValue(typeInfo)!;
+        return (Type)typeof(JsonTypeInfo).GetProperty("ElementType", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)?.GetValue(typeInfo)!;
```

Comment:
This fixes a crash in .NET 9 caused by `JsonTypeInfo.ElementType` being made public. We should make sure all copies of `JsonSchemaMapper` have ingested this change before .NET 9 is shipped.

cc @stephentoub @NTaylorMullen

Reply:
Link to SK fix: https://github.com/microsoft/semantic-kernel/pull/7611